### PR TITLE
Azure pipelines

### PR DESCRIPTION
**Pull request by miguelarias-nf**
After merge this branch into master, I need to review the pipeline configs in AzureDevOps to use the pipeline file under master branch instead of azure-pipelines

### CODE REVIEW

File: AzureDevOps.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+```bash
+    #name   = "titus-backend-${random_string.suffix.result}"
+    image  = "microsoft/aci-helloworld:latest"
+```
+
+Change the TF variables to use your own resource group id, name and location and Apply the TF plan.
+
+```bash
+# Get this information from your Azure account manager
+resource_group_id = "7d149fba-cc81-4bab-8461-ea09919d78cd"
```
Comment (NeoTech) [inline AzureDevOps.md:31]: Shouldnt commit this - as its a template.

File: infra/azure/terraform/network.tf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    address_space = ["10.0.0.0/16"]
+    location = var.location
+    resource_group_name = var.resource_group_name
+} 
+
+resource "azurerm_subnet" "titus-subnet" {
+  name                 = "titus-subnet"
+  resource_group_name  = var.resource_group_name
+  virtual_network_name = azurerm_virtual_network.titus-vnet.name
+  address_prefixes     = ["10.0.1.0/24"]
```
Comment (NeoTech) [inline infra/azure/terraform/network.tf:13]: Drive the value of the network from Variables, allowing for better control.

File: infra/azure/terraform/network.tf [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# create virtual network
+resource "azurerm_virtual_network" "titus-vnet" {
+    name = "titus-vnet"
+    address_space = ["10.0.0.0/16"]
```
Comment (NeoTech) [inline infra/azure/terraform/network.tf:4]: Drive the value of the network from Variables, allowing for better control.

**Comment by NeoTech** [comment]
Would also be nice to see DNS control in this terraform setup so it can automatically map public IPs and endpoints to FQDN, using DNS Zone.

**Review by NeoTech — COMMENTED** [review]
Looks good, just few comments, would also want to see DNS control in here.

File: AzureDevOps.md (hunk 1 above)
Comment (miguelarias-nf) [reply AzureDevOps.md]: Masked the value and documentation moved and formatted to match GCP docs

File: infra/azure/terraform/network.tf (hunk 3 above)
Comment (miguelarias-nf) [reply infra/azure/terraform/network.tf]: Done, vnet and subnet ranges defined as variables now

File: infra/azure/terraform/network.tf (hunk 2 above)
Comment (miguelarias-nf) [reply infra/azure/terraform/network.tf]: Done, vnet and subnet ranges defined as variables now

File: docs/devops/azure/_sidebar.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        - [Deploy on AWS with Mira](https://github.com/nearform/titus/tree/master/packages/titus-infra-aws-mira)
+        - [Deploy Titus on Azure](/devops/azure/?id=Deploy-Titus-on-Azure)
+            - [Requirements](/devops/azure/?id=requirements)
+            - [Infrastructure Stack](/devops/azure/?id=infrastructure-stack)
+            - [Terraform and Azure account](/devops/azure/?id=terraform-and-azure-account)
+            - [Provision Infrastructure with Terraform](/devops/azure/?id=provision-infrastructure-with-terraform)
+            - [Create the pipelines in Azure DevOps](/devops/azure/?id=create-the-pipelines-in-azure-devops)
+            - [Update the Container images](/devops/azure/?id=update-the-container-images)
+            - [Verifying Azure deployment](/devops/azure/?id=verifying-azure-deployment)
+- [Contribute to Titus](/contributing/)
```
[the remark this answers is not in the document]
Comment (miguelarias-nf) [reply docs/devops/azure/_sidebar.md:17]: Maybe I am wrong, but I think each webpage requires a sidebar with a submenu.
This is the submenu for the Azure one, GCP has it's own submenu on the sidebar which is a bit different than this one.

**Review by simoneb — APPROVED** [review]
If all feedback is addressed and we're happy about the shape of this we can:

- merge the PR
- turn off auto deployment to azure as we want to keep only 1 cloud service to which we automatically deploy (including removing the hooks) created by azure
- tear down the azure resources we're using

**Pull request merged**